### ISSUE TITLE
Rendering fails when a pattern has an UTF-8 character in a config

### ISSUE THREAD

**Issue by thibaudcolas** [torchbox/django-pattern-library#26]
In GitLab by @m1kola on Mar 19, 2019, 15:12

With `django-pattern-library==0.2.0` rendering of patterns that contain UTF-8 chars fails with the following traceback. Most likely it happens because we do not specify encoding when read a config into a string [here](https://git.torchbox.com/tech/django-pattern-library/blob/465cb3a668b9e7b007657f9f6201236ff33239ab/pattern_library/utils.py#L95).

```python
Traceback:

File "/home/vagrant/.virtualenvs/my_project/lib/python3.6/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/home/vagrant/.virtualenvs/my_project/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  126.                 response = self.process_exception_by_middleware(e, request)

File "/home/vagrant/.virtualenvs/my_project/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  124.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/home/vagrant/.virtualenvs/my_project/lib/python3.6/site-packages/django/views/generic/base.py" in view
  68.             return self.dispatch(request, *args, **kwargs)

File "/home/vagrant/.virtualenvs/my_project/lib/python3.6/site-packages/django/views/generic/base.py" in dispatch
  88.         return handler(request, *args, **kwargs)

File "/home/vagrant/.virtualenvs/my_project/lib/python3.6/site-packages/pattern_library/views.py" in get
  36.         templates = get_pattern_templates(available_pattern_types)

File "/home/vagrant/.virtualenvs/my_project/lib/python3.6/site-packages/pattern_library/utils.py" in get_pattern_templates
  76.                         pattern_config = get_pattern_config(template.origin.template_name)

File "/home/vagrant/.virtualenvs/my_project/lib/python3.6/site-packages/pattern_library/utils.py" in get_pattern_config
  101.     config_str = get_pattern_config_str(template_name)

File "/home/vagrant/.virtualenvs/my_project/lib/python3.6/site-packages/pattern_library/utils.py" in get_pattern_config_str
  95.             return str(f.read())

File "/home/vagrant/.virtualenvs/my_project/lib/python3.6/encodings/ascii.py" in decode
  26.         return codecs.ascii_decode(input, self.errors)[0]

Exception Type: UnicodeDecodeError at /pattern-library/
Exception Value: 'ascii' codec can't decode byte 0xe2 in position 210: ordinal not in range(128)

```

**Comment by thibaudcolas** [comment]
In GitLab by @m1kola on Mar 19, 2019, 15:23

mentioned in commit a8734a2cf6130c36c2c09d82af8d8a3ff2bd1204

**Comment by thibaudcolas** [comment]
In GitLab by @m1kola on Mar 19, 2019, 16:25

mentioned in commit 3ab4864146acd76ec296bee4a823c91da49ccd3c

**Comment by thibaudcolas** [comment]
In GitLab by @m1kola on Mar 19, 2019, 16:37

closed via commit 3ab4864146acd76ec296bee4a823c91da49ccd3c

**Comment by thibaudcolas** [comment]
In GitLab by @m1kola on Mar 19, 2019, 16:37

closed via merge request !37
